src/main/java/no/nav/pto/veilarbportefolje/aktiviteter/AktiviteterRepositoryV2.java

```diff
@@ -56,6 +74,82 @@ DO UPDATE SET (AKTOERID, AKTIVITETTYPE, AVTALT, FRADATO, TILDATO, STATUS, VERSIO
         );
     }
 
+    public void deleteById(String aktivitetid) {
+        log.info("Sletter alle aktiviteter med id {}", aktivitetid);
+        db.update(String.format("DELETE FROM %s WHERE %s = ?", TABLE_NAME, AKTIVITETID), aktivitetid);
+    }
+
+    public void leggTilAktiviteterFraAktivitetsplanen(String aktoerIder, boolean avtalt, HashMap<AktorId, List<AktivitetEntityDto>> result) {
+        var params = new MapSqlParameterSource();
+        params.addValue("ikkestatuser", aktivitetsplanenIkkeAktiveStatuser);
+        params.addValue("avtalt", avtalt);
+        params.addValue("ids", aktoerIder);
+        namedDb.query("""
+                        SELECT aktoerid, tildato, fradato, aktivitettype FROM aktiviteter
+                        WHERE avtalt = :avtalt::boolean AND NOT status = ANY (:ikkestatuser::varchar[]) AND aktoerid = ANY (:ids::varchar[])
+                        """,
+                params, (ResultSet rs) -> {
+                    while (rs.next()) {
+                        AktorId aktoerId = AktorId.of(rs.getString("aktoerid"));
+                        mapAktivitetTilEntity(rs).ifPresent(aktivitet ->
+                                leggTilAktivitetPaResultat(aktoerId, aktivitet, result)
```

Comment:
i dont like this part, its bit unreadable, and complicated. Why you dont populate result here, instead of passing it as argument...